.github/workflows/luarocks.yml

```diff
@@ -0,0 +1,42 @@
+name: Luarocks
+
+on:
+  push:
+    tags:
+      - '*'
+  release:
+    types:
+      - created
+  pull_request: # Tests packaging on PR
+  workflow_dispatch:
+
+jobs:
+  luarocks-upload:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Required to get the tags
+      - name: Get Version
+        run: echo "LUAROCKS_VERSION=$(git describe --abbrev=0 --tags)" >> $GITHUB_ENV
+
+      - name: Install C/C++ Compiler
+        uses: rlalik/setup-cpp-compiler@master
+        with:
+          compiler: clang-latest
+
+      - name: Install tree-sitter CLI
+        uses: baptiste0928/cargo-install@v3
+        with:
+          crate: tree-sitter-cli
```

Comment:
I assume both of these are needed for the `tree-sitter-*` dependencies but I can't quite figure out why. 

Like why does the environment that publishes the package need to have `tree-sitter-cli` available?

Do the dependencies need to be built as part of the upload?

Reply:
luarocks-tag-release tries to install the package from the generated rockspec, before trying to upload (which also installs and builds dependencies).
If installing fails, it cancels the upload.

Reply:
Ah, okay, that makes sense, to make sure what you upload  actually works. I also now get what the `Tests packaging on PR` comment means. Thanks!